src/main/java/life/qbic/model/download/QbicDataDownloader.java

```diff
@@ -187,134 +193,185 @@ private void downloadDatasets(Sample analyte, List<DataSet> dataSets,
     }
+    return downloadResponse;
   }
 
-  private void downloadFile(DataSetFile dataSetFile, Path prefix) throws DownloadException {
+  private FileDownloadResponse downloadFile(DataSetFile dataSetFile, Path prefix) throws DownloadException {
     for (IDataStoreServerApi dataStoreServer : dataStoreServers) {
       try {
-        downloadFileFromDataStoreServer(dataSetFile, prefix, dataStoreServer);
-        break;
+        return downloadFileFromDataStoreServer(dataSetFile, prefix, dataStoreServer);
       } catch (FileNotFoundException fileNotFoundException) {
         // log and try the next data store server
         LOG.debug("Download attempt failed on " + dataStoreServer, fileNotFoundException);
-        continue;
       }
     }
+    return FileDownloadResponse.failure();
   }
 
+  // note: DataSetFileDownloadReader closes the input stream after it finished reading it.
+  private static class AutoClosableDataSetFileDownloadReader extends DataSetFileDownloadReader implements AutoCloseable {
+    public AutoClosableDataSetFileDownloadReader(InputStream in) {
+      super(in);
+    }
+  }
+
+  public static class FileDownloadResponse {
+    private final boolean success;
+
+    private FileDownloadResponse(boolean success) {
+      this.success = success;
+    }
+
+    public static FileDownloadResponse success() {
+      return new FileDownloadResponse(true);
+    }
+
+    private static FileDownloadResponse failure() {
+      return new FileDownloadResponse(false);
+    }
+
+    public boolean isSuccess() {
+      return success;
+    }
+
+    public boolean isFailure() {
+      return !success;
+    }
+  }
+
+
   /**
-   * @throws FileNotFoundException if the file was not found on the data store server.
+   * @throws FileNotFoundException if the file was not on the data store server.
    */
-  private void downloadFileFromDataStoreServer(DataSetFile dataSetFile, Path prefix,
+  private FileDownloadResponse downloadFileFromDataStoreServer(DataSetFile dataSetFile, Path prefix,
       IDataStoreServerApi dataStoreServerApi) throws FileNotFoundException {
 
+    // skip if file is empty
+    if (dataSetFile.getFileLength() < 1) {
+      LOG.info("Skipped empty file " + dataSetFile.getPath());
+      return FileDownloadResponse.success();
+    }
+
+    Path localFilePath = OutputPathFinder.determineOutputDirectory(outputPath, prefix, dataSetFile,
+        conservePaths);
+    File localFile = localFilePath.toFile();
+
+    // skip if same content exists already
+    if (localFile.exists()) {
+      System.out.print("Found existing file. Checking content...");
+
+      if (localFileWithSameContent(dataSetFile, localFile)) {
+        System.out.println("\rFound existing file with identical content. Skipping "
+            + localFile.getAbsolutePath());
+        return FileDownloadResponse.success();
+      } else {
+        System.out.println("\rUpdating existing file " + localFile.getAbsolutePath());
```

Comment:
Logging?

Reply:
This is only for the user. The log contains debug information on files found with crc32 checksum in hex format.
```
  private boolean localFileWithSameContent(DataSetFile dataSetFile, File localFile) {
    ChecksumValidationResult checksumValidationResult = checksumForFile(dataSetFile, localFile);
    if (checksumValidationResult.isValid()) {
      LOG.debug(checksumValidationResult.computedChecksum + " " + localFile.getName()
          + " exists locally.");
      return true;
    }
    return false;
  }
```

Reply:
```
2023-01-23 12:26:15.177 [main] INFO  life.qbic.model.download.QbicDataDownloader - Found 1 dataset(s) for sample ***
2023-01-23 12:26:15.344 [main] DEBUG life.qbic.model.download.QbicDataFinder - 6 files and directories found in dataset *** on dss HTTP invoker proxy for service URL [***]
2023-01-23 12:26:15.472 [main] DEBUG life.qbic.model.download.QbicDataFinder - No files found in dataset *** on dss HTTP invoker proxy for service URL [***]
2023-01-23 12:26:15.473 [main] INFO  life.qbic.model.download.QbicDataDownloader - Downloading 3 files for dataset *** (***)
2023-01-23 12:26:15.501 [main] DEBUG life.qbic.model.download.QbicDataDownloader - 70fe15f2 ***.CEL exists locally.
2023-01-23 12:26:15.502 [main] DEBUG life.qbic.model.download.QbicDataDownloader - 258ab72b ***.CEL.origlabfilename exists locally.
2023-01-23 12:26:15.504 [main] DEBUG life.qbic.model.download.QbicDataDownloader - 1ea50c8f ***.CEL.sha256sum exists locally.
```